CMakeLists.txt

```diff
@@ -0,0 +1,962 @@
+        ${GSRC}/src/optimizer/eigenTorques.cpp)
+
+    set (OPTIMIZER_HEADERS 
+        ${OPTIMIZER_HEADERS} 
+        ${GSRC}/src/optimizer/eigenTorques.h
+)
+endif (CGDB)    
+
+set (MOC_HEADERS
+    ${MOC_HEADERS}
+    ${GSRC}/src/optimizer/optimizerDlg.h
+)
+
+
+
+##########################################
+## Generate Qt files for the icon sets  ## 
+##########################################
+
+# Custom command to create an image collection with uic3 (it has to be uic3, not uic) like this: 
+# uic3 -embed graspit -o qt_image_collection.cpp path_to_my_images/*
+# The qt_image_collection.cpp must then be added to the sources to compile.
+# I also tried adding images as resource (QT4_ADD_RESOURCES) but this didn't work with the 
+# current GraspIt source.
+ADD_CUSTOM_COMMAND(OUTPUT 
+    ${CMAKE_CURRENT_BINARY_DIR}/qt_image_collection.cpp 
+    COMMAND ${QT_UIC3_EXECUTABLE} 
+    ARGS -o ${CMAKE_CURRENT_BINARY_DIR}/qt_image_collection.cpp 
+        -embed graspit ${GRASPIT_IMAGES} 
+    DEPENDS ${GRASPIT_IMAGES})
+
+SET(GRASPIT_IMAGE_SOURCES 
+    ${CMAKE_CURRENT_BINARY_DIR}/qt_image_collection.cpp)
+
+################################
+## Make Qt Moc files of GraspIT
+################################
+
+set (ALL_GRASPIT_FORMS
+    ${GRASPIT_FORMS}
+    ${CGDB_FORMS}
+    ${OPTIMIZER_FORMS}
+)
+
+set (GRASPIT_UI_INCLUDE_DIRNAME ui)
+
+# we will need to create a special folder named GRASPIT_UI_INCLUDE_DIRNAME
+# into which to generate the UI files, so that they can be included from
+# the graspit header files with the right prefix.
+
+set (GRASPIT_UI_INCLUDE_PATH ${CMAKE_CURRENT_BINARY_DIR}/include/${PROJECT_NAME}/${GRASPIT_UI_INCLUDE_DIRNAME})
+
+    
+file(MAKE_DIRECTORY ${GRASPIT_UI_INCLUDE_PATH})
+
+set (OLD_CMAKE_CURRENT_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}) # backup current binary dir
+set (CMAKE_CURRENT_BINARY_DIR ${GRASPIT_UI_INCLUDE_PATH}) #artificially changing output location for UI files
+
+QT4_WRAP_UI(UI_OUTFILES ${ALL_GRASPIT_FORMS}) # generate UI outfiles (will use CMAKE_CURRENT_BINARY_DIR)
+
+set (CMAKE_CURRENT_BINARY_DIR ${OLD_CMAKE_CURRENT_BINARY_DIR}) # restore current binary dir
+
+QT4_WRAP_CPP(MOC_OUTFILES ${MOC_HEADERS})
+
+
+# add Qt stuff to the includes
+set (GRASPIT_INCLUDES 
+     ${GRASPIT_INCLUDES}
+     ${GRASPIT_UI_INCLUDE_PATH}
+     ${SOQT_INCLUDE_DIRS}
+     ${QT_INCLUDES}
+     ${QT_INCLUDE_DIR}
+     ${QT_QT3SUPPORT_INCLUDE_DIR}
+     ${QHULL_INCLUDE_DIRS}
+)
+
+
+## Specify additional locations of header files
+include_directories(
+    ${GRASPIT_INCLUDES}
+    ${TINYXML_INCLUDES}
+)
+
+
+############
+## Build  ## 
+############
+
+add_definitions(${GRASPIT_DEFINES} -g)
+
+set (GRASPIT_LIBRARY_SOURCES
+    ${GRASPIT_CORE_SOURCES}
+    ${CGDB_SOURCES}
+    ${PLY_SOURCES}
+    ${GRASPIT_COLLISION_SOURCES}
+    ${GRASPIT_DYNAMICS_SOURCES}
+    ${GRASPIT_OPTIMIZER_SOURCES}
+    ${GRASPIT_FORM_SOURCES}
+    ${GRASPIT_IMAGE_SOURCES}
+    ${UI_OUTFILES}
+    ${MOC_OUTFILES}
+)
+
+if (NOT SKIP_SHARED_LIB)
```

Comment:
You are right, in the future it makes a lot of sense to build most of graspit as a lib, with the UI as an executable linking against it (or the command prompt version, etc.)

For now though, especially since this is a big change, I would prefer to keep things as simple as possible. If you don't mind maintaining your own slightly modified CMakeLists in your fork, I would prefer to go with a single build option...